src/contexts/ErrorContext.jsx

```diff
@@ -0,0 +1,24 @@
+// src/contexts/ErrorContext.js
+import React, { createContext, useState, useContext } from 'react';
+
+const ErrorContext = createContext();
+
+export const useError = () => useContext(ErrorContext);
+
+export const ErrorProvider = ({ children }) => {
+  const [error, setError] = useState(null);
+
+  const showError = (message) => {
```

Comment:
no need to have two functions for the purpose of changing state for `error`, formulate a single function `toggleErrorState` which can take a parameter based on which this value can be changed accordingly